src/simplewallet/simplewallet.cpp

```diff
@@ -588,11 +590,21 @@ bool simple_wallet::print_seed(bool encrypted)
 
 bool simple_wallet::seed(const std::vector<std::string> &args/* = std::vector<std::string>()*/)
 {
+  if (m_wallet->key_on_device())
```

Comment:
> viewkey and spendkey are supported but display 00---00 and FF...FF for private view/spend keys.

These commands are supposed to print private keys, which aren't available here. Printing 00 and FF will confuse users, IMO.

> `seed_set_language` does not have any effect

Again, seed language is relevant only when the wallet knows the spend secret key (i.e. the seed is just another representation of the spend secret key). Basically, the test `if (m_wallet->key_on_device())` should be added always where these tests `if (m_wallet->multisig())` and `if (m_wallet->watch_only())` are used.

> `submit_transfer`: not sure about what this command do

This command needs to decrypt an unsigned transaction file that was encrypted using the view secret key. Because the view key is stored only in the hardware, it will not work.

> `sign` should work.

Pretty sure it won't work. Again, this is where the tests `if (m_wallet->watch_only())` and `if (m_wallet->multisig())` are present, meaning that the function requires the spend secret key. Since it only exists in the ledger device, it won't work.

Reply:
Ok, let's display a specific message for secret key with spendkey/viewkey commands

Avoid other commands.